### PR TITLE
ESLint incorporated into the project

### PR DESCRIPTION
**Pull request by ashwin713**
ESLint rules updated to improve the coding standard.

### CODE REVIEW

**Review by sreeragh-ar — CHANGES_REQUESTED** [review]
Ensure arrow functions do not omit parentheses(when they have only one parameter). They might look like conditional statements.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      "error",
+      "never"
+    ],
+    "no-console": [
+      "error",
+      {"allow": ["warn", "error"]}
+    ],
+    "curly": [
+      "error"
+    ],
```
Comment (surajdevesan) [inline .eslintrc.json:94]: Please confirm if this will be an issue for arrow functions? As @SragAR pointed out.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      {"argsIgnorePattern": "^__"}
+    ],
+    "keyword-spacing": [
+      "error",
+      {"before": true,"after": true}
+    ],
+    "space-before-blocks": [
+      "error"
+    ]
+  }
```
Comment (surajdevesan) [inline .eslintrc.json]: Please review these rules and include if possible
 - camelcase rule
- strict
- no-tabs

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      "unix"
+    ],
+    "operator-linebreak": [
+      "error",
+      "after"
+    ],
+    "quotes": [
+      "error",
+      "single",
+      {"avoidEscape": true}
```
Comment (surajdevesan) [inline .eslintrc.json:46]: will this work with backticks?

**Review by surajdevesan — CHANGES_REQUESTED** [review]
Please add a description to the PR.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        "es6": true,
+        "node": true
+    },
+    "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:import/errors",
+        "plugin:import/warnings"
+    ],
+    "parserOptions": {
```
Comment (thatsmeamal) [inline .eslintrc.json:13]: You may add the 'strict' configuration here.

File: .eslintrc.json (hunk 2 above)
Comment (thatsmeamal) [reply .eslintrc.json]: @surajdevesan 'no-tabs', it is implied in the indent rule :stuck_out_tongue_winking_eye:

Comment (surajdevesan) [reply .eslintrc.json]: @thatsmeamal no-tabs rule ensures that **no tab** characters are there in the code including comments and other operations 
ex:  ` var a /t= 2;` *which is only caught by no-tab rule and not indent rule.*
**Please refer the [docs](https://eslint.org/docs/rules/no-tabs#rule-details) for further clarification** :stuck_out_tongue_closed_eyes:

**Review by qb-sreeshma — COMMENTED** [review]
Please check whether the following rules should be included for better coding conventions

"brace-style": [2, "1tbs"]
"space-in-brackets": [2, "never"],
"space-after-keywords": [2, "always"],
"eqeqeq": [2, "smart"]

File: .eslintrc.json (hunk 4 above)
Comment (thatsmeamal) [reply .eslintrc.json]: '[;;';'

**Review by thatsmeamal — CHANGES_REQUESTED** [review]
Please make necessary changes.

File: .eslintrc.json (hunk 3 above)
Comment (ashwin713) [reply .eslintrc.json:46]: https://eslint.org/docs/rules/quotes
go through this, i think this would suffice.

File: .eslintrc.json (hunk 1 above)
Comment (ashwin713) [reply .eslintrc.json:94]: updated the rule set

File: .eslintrc.json (hunk 2 above)
Comment (ashwin713) [reply .eslintrc.json]: updated the rule set

File: .eslintrc.json (hunk 4 above)
Comment (ashwin713) [reply .eslintrc.json:13]: updated the rule set

File: .eslintrc.json (hunk 3 above)
Comment (surajdevesan) [reply .eslintrc.json:46]: (y)

**Review by surajdevesan — APPROVED** [review]
:shipit:

**Review by princelonappan07 — APPROVED** [review]

**Review by vyshakn — APPROVED** [review]
:shipit:

**Review by vyshakn — APPROVED** [review]

**Review by qb-sreeshma — APPROVED** [review]

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      "error",
+      {"before": true,"after": true}
+    ],
+    "space-before-blocks": [
+      "error"
+    ],
+    "camelcase": "error",
+    "no-tabs": "error",
+    "eqeqeq": ["error", "smart"],
+    "brace-style": ["error", "1tbs", { "allowSingleLine": true }],
```
Comment (thatsmeamal) [inline .eslintrc.json:109]: Don't you think 'allowSingleLine' option can tamper with the look of the code ?

**Review by thatsmeamal — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 5 above)
Comment (ashwin713) [reply .eslintrc.json:109]: This rule allows the code such as : 
```javascript
if (foo) { bar(); }
```

Comment (ashwin713) [reply .eslintrc.json:109]: its fine right ?

Comment (thatsmeamal) [reply .eslintrc.json:109]: OK, then

**Review by thatsmeamal — APPROVED** [review]

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    ],
+    "linebreak-style": [
+      "error",
+      "unix"
+    ],
+    "operator-linebreak": [
+      "error",
+      "after"
+    ],
+    "quotes": [
```
Comment (sreeragh-ar) [inline .eslintrc.json:43]: Error running this configuration.
Configuration for rule "quotes" is invalid: Value "single,[object Object],[object Object]" should NOT have more than 2 items.

**Comment by sreeragh-ar** [comment]
Add config to ensure space  around infix operators.

**Review by sreeragh-ar — APPROVED** [review]

**Review by Sanandanan — APPROVED** [review]

**Pull request merged**